e2e/cypress/integration/system_console/team_and_channel_assign_roles_spec.js

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('System Console', () => {
+    it('MM-20059 - System Admin can map roles to groups from Team Configuration screen', () => {
+        cy.apiLogin('sysadmin');
+
+        // # Go to system admin page and to team configuration page of channel "eligendi"
+        cy.visit('/admin_console/user_management/teams');
+        cy.findByTestId('eligendiedit').click();
+
+        // # Wait until the groups retrieved and show up
+        cy.wait(500); //eslint-disable-line cypress/no-unnecessary-waiting
+
+        // # Remove all exisiting groups
+        cy.get('#groups-list--body').then((el) => {
+            if (el[0].childNodes[0].innerText !== 'No groups specified yet') {
+                for (let i = 0; i < el[0].childNodes.length; i++) {
+                    cy.get('#group-actions').click();
+                }
+
+                // # Save the setting
+                cy.get('#saveSetting').click();
+            }
+        });
+
+        // # Add the first group in the group list then save
+        cy.findByTestId('add-group').click();
+        cy.get('#multiSelectList').first().click();
+        cy.get('#saveItems').click();
+        cy.get('#saveSetting').click();
+
+        // # Change the role from member to Channel Admin
+        cy.findByTestId('current-role').click();
+        cy.get('#role-to-be').click();
+        cy.get('#saveSetting').click();
+
+        // # Reload to ensure it's been saved
+        cy.reload();
+
+        // * Check to make the the current role text is displayed as Team Admin
+        cy.findByTestId('current-role').should('have.text', 'Team Admin');
```

Comment:
Can we get the current Team Role and then Swap the role name by selecting the new role. The existing row might already have Team Admin selected. 
Also please add an assertion to verify that only one option is shown when the Roles dropdown is opened.

Reply:
Done! I have added code to the test to ensure that only one option is available (Admin or Member)